wagtail/snippets/tests/test_viewset.py

```diff
@@ -1493,3 +1494,40 @@ def test_inspect_view(self):
             {"url": "", "label": "Inspect"},
         ]
         self.assertBreadcrumbsItemsRendered(items, response.content)
+
+
+class TestSnippetSearch(BaseSnippetViewSetTests):
+    model = SnippetSearchTestModel
+
+    @classmethod
+    def setUpTestData(cls):
+        # make three objects with different titles and body
+        objects = [
+            cls.model(title="Hello World", body="This one is classic"),
+            cls.model(title="Hello Anime", body="We all love anime(opinions may vary)"),
+            cls.model(title="Food", body="I like food, do you?"),
+        ]
+        cls.model.objects.bulk_create(objects)
+
+    # write test for single post search,multiple post search and no post search
+
+    def test_single_post_search(self):
+        response = self.client.get(self.get_url("list"), {"q": "Hello World"})
```

Comment:
I think rather than having separate cases for whether the result is single vs. multiple, it's more useful to test based on different fields, e.g. have this one test the `body` filtering.